live/app/components/signup-form.js

```diff
@@ -0,0 +1,148 @@
+}
+
+function getValidationStatus(isValidField) {
+  return (isValidField) ? 'error' : 'success';
+}
+
+function isGivenNotEmptyValue(value) {
+  return value.trim() ? true : false;
+}
+
+export default Ember.Component.extend({
+  classNames: ['signup-form'],
+  registrationMessage: '',
+  validation: {
+    lastName: {
+      message: null,
+      status: 'default'
+    },
+    firstName: {
+      status: 'default',
+      message: null
+    },
+    email: {
+      status: 'default',
+      message: null
+    },
+    password: {
+      status: 'default',
+      message: null
+    },
+    cgu: {
+      status: 'default',
+      message: null
+    }
+  },
+  temporaryAlert: {
+    status:'default',
+    message: ''
+  },
+
+  _updateTextfieldModelObject (key, status, message) {
+    const statusObject = 'validation.'+key+'.status';
+    const messageObject = 'validation.'+key+'.message';
+    this.set(statusObject, status);
+    this.set(messageObject, message);
+  },
+
+  _getModelAttributeValueFromKey(key) {
+    const userModel = this.get('user');
+    return userModel.get(key);
+  },
+
+  _toggleConfirmation(status, message) {
+    this.set('temporaryAlert', {status: TEMPORARY_DIV_CLASS_MAP[status], message});
+    if(config.APP.isMessageStatusTogglingEnabled) {
+      Ember.run.later(() => {
+        this.set('temporaryAlert', {status: 'default', message: ''});
+      }, config.APP.MESSAGE_DISPLAY_DURATION);
+    }
+  },
+
+  _resetValidationFields(){
+    const defaultValidationObject = {
+      lastName: {
+        message: null,
+        status: 'default'
+      },
+      firstName: {
+        status: 'default',
+        message: null
+      },
+      email: {
+        status: 'default',
+        message: null
+      },
+      password: {
+        status: 'default',
+        message: null
+      },
+      cgu:{
+        status: 'default',
+        message: null
+      }
+    };
+    this.set('validation', defaultValidationObject);
+  },
+
+  _updateInputsStatus(){
+    const errors =  this.get('user.errors.content');
+    errors.forEach(({ attribute, message }) =>{
+      this._updateTextfieldModelObject(attribute, 'error', message);
+    });
+  },
+
+  _executeFieldValidation: function (key, isValid) {
+    const modelAttrValue = this._getModelAttributeValueFromKey(key);
+    const isValidInput = !isValid(modelAttrValue);
+    const status = getValidationStatus(isValidInput);
+    const message = getErrorMessage(status, key);
+    this._updateTextfieldModelObject(key, status, message);
+  },
+
+  actions: {
+    validateInput(key){
+      this._executeFieldValidation(key, isGivenNotEmptyValue);
+    },
+
+    validateInputEmail(key){
+      this._executeFieldValidation(key, isEmailValid);
+    },
+
+    validateInputPassword(key){
+      this._executeFieldValidation(key, isPasswordValid);
+    },
+
+    signup(){
+      const userObject = this.get('user');
+      userObject.save()
```

Comment:
En l'occurence j'ai l'impression que juste `this.get('user').save() …` serait un usage raisonnablement compact et lisible : le bénéfice de la variable intermédiaire me semble faible ici.